components/compositing/constellation.rs

```diff
@@ -1869,12 +1869,17 @@ impl<LTF: LayoutThreadFactory, STF: ScriptThreadFactory> Constellation<LTF, STF>
         if let Some((ref mut rng, _)) = self.random_pipeline_closure {
             if let Some(pipeline_id) = rng.choose(&*pipeline_ids) {
                 if let Some(pipeline) = self.pipelines.get(pipeline_id) {
-                    // Don't kill the root pipeline
-                    if pipeline.parent_info.is_none() { return; }
-                    // Note that we deliberately do not do any of the tidying up
-                    // associated with closing a pipeline. The constellation should cope!
-                    info!("Randomly closing pipeline {}.", pipeline_id);
-                    pipeline.force_exit();
+                    // Don't kill the mozbrowser pipeline
+                    if prefs::get_pref("dom.mozbrowser.enabled").as_boolean().unwrap_or(false) &&
+                        pipeline.parent_info.is_some()
```

Comment:
Oops, I flip-flopped on that if test, and ended up flopping where I should have flipped.